### PR TITLE
fix: Support custom g type name

### PR DESCRIPTION
**Pull request by Tanyuu**
fix: #329

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sagilio

File: Casbin.UnitTests/ModelTests/EnforcerTest.cs [hunk 1]
```diff
@@ -589,16 +589,16 @@ public void TestMultipleGroupTypeModelInMemory()
         IModel m = DefaultModel.Create();
         m.AddDef("r", "r", "sub, obj, act");
         m.AddDef("p", "p", "sub, obj, act");
-        m.AddDef("g", "g", "_, _");
+        m.AddDef("g", "g1", "_, _");
```
Comment (sagilio) [inline Casbin.UnitTests/ModelTests/EnforcerTest.cs:592]: Plz add new test for the issue case

**Review by sagilio — CHANGES_REQUESTED** [review]

File: Casbin/Model/DefaultSections.cs [hunk 2]
```diff
@@ -85,12 +85,13 @@ public bool AddSection(string section, string key, string value)
         Dictionary<string, IReadOnlyAssertion> assertionMap = new() { [key] = assertion };
 
         // TryAdd is not supported in .NET 4.5
-        if (_assertionsMap.ContainsKey(key))
+        if (_assertionsMap.ContainsKey(key) ||
```
Comment (sagilio) [inline Casbin/Model/DefaultSections.cs:88]: Why not use `_assertionsMap.ContainsKey(section)`

File: Casbin.UnitTests/ModelTests/EnforcerTest.cs (hunk 1 above)
Comment (hsluoyz) [reply Casbin.UnitTests/ModelTests/EnforcerTest.cs:617]: @Tanyuu reply to all review comments and resolve if needed

File: Casbin/Model/DefaultSections.cs (hunk 2 above)
Comment (hsluoyz) [reply Casbin/Model/DefaultSections.cs:88]: @Tanyuu reply to all review comments and resolve if needed

**Comment by hsluoyz** [comment]
@sagilio

File: Casbin.UnitTests/ModelTests/EnforcerTest.cs [hunk 3]
```diff
@@ -608,22 +608,47 @@ public void TestMultipleGroupTypeModelInMemory()
...
+    }
+
     [Fact]
     public async Task TestMultipleGroupTypeModelInMemoryAsync()
     {
         IModel m = DefaultModel.Create();
         m.AddDef("r", "r", "sub, obj, act");
         m.AddDef("p", "p", "sub, obj, act");
-        m.AddDef("g", "g", "_, _");
+        m.AddDef("g", "g1", "_, _");
```
Comment (sagilio) [inline Casbin.UnitTests/ModelTests/EnforcerTest.cs:642]: Plz reset here

File: Casbin.UnitTests/ModelTests/EnforcerTest.cs (hunk 1 above)
Comment (sagilio) [reply Casbin.UnitTests/ModelTests/EnforcerTest.cs:617]: @Tanyuu Maybe we can use the test case from the user at https://github.com/casbin/Casbin.NET/issues/329#issue-1851384512?

**Review by sagilio — CHANGES_REQUESTED** [review]

**Review by sagilio — APPROVED** [review]
LGTM

**Pull request merged**